src/Library/demos/File Dialog/main.js

```diff
@@ -0,0 +1,125 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+
+Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
+Gio._promisify(
+  Gtk.FileDialog.prototype,
+  "open_multiple",
+  "open_multiple_finish",
+);
+Gio._promisify(Gtk.FileDialog.prototype, "save", "save_finish");
+Gio._promisify(
+  Gtk.FileDialog.prototype,
+  "select_folder",
+  "select_folder_finish",
+);
+Gio._promisify(
+  Gtk.FileDialog.prototype,
+  "select_multiple_folders",
+  "select_multiple_folders_finish",
+);
+
+const button = workbench.builder.get_object("button");
+const select_mode = workbench.builder.get_object("select_mode");
+
+let selected_mode = select_mode.selected_item.get_string();
+select_mode.connect("notify::selected-item", () => {
+  selected_mode = select_mode.selected_item.get_string();
+  console.log(`Mode: ${selected_mode}`);
+});
+
+const dialog_for_file = new Gtk.FileDialog({
+  title: _("Select File(s)"),
+  modal: true,
+});
+
+const dialog_for_folder = new Gtk.FileDialog({
+  title: _("Select Folder(s)"),
+  modal: true,
+});
+
+const dialog_for_save = new Gtk.FileDialog({
+  title: _("Save File"),
+  modal: true,
+});
+
+button.connect("clicked", () => {
+  switch (selected_mode) {
+    case "Select File":
+      open_file().catch(logError);
+      break;
+    case "Select File (Folder pre-defined)":
+      open_file_folder().catch(logError);
+      break;
```

Comment:
I don't think we need both. "Select File" with a pre-defined folder is a good example. If for some reason the user doesn't want a pre-defined folder, it is easy to adjust the code.